.github/workflows/ci-core.yml

```diff
@@ -340,23 +340,50 @@ jobs:
           fetch-depth: 0 # fetches all history for all tags and branches to provide more metadata for sonar reports
       - name: Download all workflow run artifacts
         uses: actions/download-artifact@c850b930e6ba138125429b7e5c93fc707a7f8427 # v4.1.4
+
       - name: Set SonarQube Report Paths
         id: sonarqube_report_paths
         shell: bash
         run: |
           echo "sonarqube_tests_report_paths=$(find go_core_tests_logs -name output.txt | paste -sd "," -)" >> $GITHUB_OUTPUT
           echo "sonarqube_coverage_report_paths=$(find go_core_tests_logs -name coverage.txt | paste -sd "," -)" >> $GITHUB_OUTPUT
+          echo "sonarqube_lint_report_paths=$(find golangci-lint-report -name golangci-lint-report.xml | paste -sd "," -)" >> $GITHUB_OUTPUT
+
+      - name: Check SonarQube Report Paths
+        id: check_sonarqube_paths
+        run: |
+          ARGS=""
+
+          if [[ -z "${{ steps.sonarqube_report_paths.outputs.sonarqube_tests_report_paths }}" ]]; then
+            echo "::warning::No test report paths found, will not pass to sonarqube"
+          else
+            ARGS="$ARGS -Dsonar.go.tests.reportPaths=${{ steps.sonarqube_report_paths.outputs.sonarqube_tests_report_paths }}"
+          fi
+
+          if [[ -z "${{ steps.sonarqube_report_paths.outputs.sonarqube_coverage_report_paths }}" ]]; then
+            echo "::warning::No coverage report paths found, will not pass to sonarqube"
+          else
+            ARGS="$ARGS -Dsonar.go.coverage.reportPaths=${{ steps.sonarqube_report_paths.outputs.sonarqube_coverage_report_paths }}"
+          fi
+
+          if [[ -z "${{ steps.sonarqube_report_paths.outputs.sonarqube_lint_report_paths }}" ]]; then
+            echo "::warning::No lint report paths found, will not pass to sonarqube"
+          else
+            ARGS="$ARGS -Dsonar.go.golangci-lint.reportPaths=${{ steps.sonarqube_report_paths.outputs.sonarqube_lint_report_paths }}"
+          fi
+
+          echo "SONARQUBE_ARGS=$ARGS" >> $GITHUB_ENV
```

Comment:
We shouldn't have any multiline output problems to deal with, right?

Reply:
I don't think so. This is a standard practice amongst many repos utilizing sonar scan.